adobe_vipm/flows/sync.py

```diff
@@ -355,8 +327,22 @@ def sync_agreement(mpt_client, agreement, dry_run):
 
     except Exception as e:
         logger.exception(f"Error synchronizing agreement {agreement["id"]}: {e}")
-        notify_agreement_unhandled_exception_in_teams(
-            agreement["id"], traceback.format_exc()
+        notify_agreement_unhandled_exception_in_teams(agreement["id"], traceback.format_exc())
+    else:
+        _update_last_sync_date(mpt_client, agreement, dry_run)
+
+
+def _update_last_sync_date(mpt_client: MPTClient, agreement: dict, dry_run: bool) -> None:
+    if not dry_run:
```

Comment:
I would recommend putting it out of the function :-) In the `else` branch. Just to have everything in one place, why this parameter can't be updated. So we have 2 reasons:
1. Dry run
2. Failure

Otherwise these 2 conditions a hidden in 2 different places